tests/lib/snaps/test-snapd-profiler-core18/profiler.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+
+import datetime
```

Comment:
I would suggest to tweak the tool like this:
```
#!/usr/bin/env python3

import logging
import time
import json
import psutil

PROCATTRS = set(['pid', 'name', 'cmdline', 'cpu_times',
                 'create_time', 'cpu_percent', 'memory_info',
                 'memory_percent'])
PROCS = set(["snap", "snapd"])
RATE = {"snap": 3, "snapd": 1}
INTERVAL = 1


def main():
    logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(message)s')

    count = 0
    while True:
        procs = [proc for proc in psutil.process_iter() if proc.name() in PROCS]
        for proc in procs:
            if count % RATE.get(proc.name(), 1) == 0:
                logging.info(json.dumps(proc.as_dict(attrs=PROCATTRS)))
        time.sleep(INTERVAL)


if __name__ == '__main__':
    main()
```

Plus the changes to log into a file. Instead of logging a dictionary of attributes (some of which may be converted to a string in an unexpected way), the output is JSON. This would enable us to apply further processing with ease.

Reply:
done